Formula/texlive.rb

```diff
@@ -0,0 +1,161 @@
+class Texlive < Formula
+  desc "Free software distribution for the TeX typesetting system"
+  homepage "https://www.tug.org/texlive/"
+  url "https://github.com/TeX-Live/texlive-source/archive/refs/tags/svn58837.tar.gz"
+  sha256 "0afa6919e44675b7afe0fa45344747afef07b6ee98eeb14ff6a2ef78f458fc12"
+  license :public_domain
+  head "https://github.com/TeX-Live/texlive-source.git", branch: "trunk"
+
+  depends_on "cairo"
+  depends_on "clisp"
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "gd"
+  depends_on "ghostscript"
+  depends_on "gmp"
+  depends_on "graphite2"
+  depends_on "harfbuzz"
+  depends_on "libpng"
+  depends_on "lua"
+  depends_on "luajit-openresty"
+  depends_on "mpfr"
+  depends_on "perl"
+  depends_on "pixman"
+  depends_on "potrace"
+
+  uses_from_macos "icu4c"
+  uses_from_macos "ncurses"
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "gcc"
+    depends_on "libice"
+    depends_on "libsm"
+    depends_on "libx11"
+    depends_on "libxaw"
+    depends_on "libxext"
+    depends_on "libxmu"
+    depends_on "libxpm"
+    depends_on "libxt"
+  end
+
+  fails_with gcc: "5"
+
+  resource("texlive-extra") do
+    url "https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2021/texlive-20210325-extra.tar.xz"
+    sha256 "46a3f385d0b30893eec6b39352135d2929ee19a0a81df2441bfcaa9f6c78339c"
+  end
+
+  resource "install-tl" do
+    url "https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2021/install-tl-unx.tar.gz"
+    sha256 "74eac0855e1e40c8db4f28b24ef354bd7263c1f76031bdc02b52156b572b7a1d"
+  end
+
+  resource("texlive-texmf") do
+    url "https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2021/texlive-20210325-texmf.tar.xz"
+    sha256 "ff12d436c23e99fb30aad55924266104356847eb0238c193e839c150d9670f1c"
+  end
+
+  def install
```

Comment:
```suggestion
  end
  
  conflicts_with "latexindent", because: "both install `latexindent` binaries"

  def install
```

Reply:
I didn't catch this one, but we might want to disable `latexindent` instead in the configuration options.  I've already disabled some other features because they conflict with existing formula and I see no reason we can't do the same here.

Reply:
ah missed that, yea that makes more sense

Reply:
I tried to see if `--disable-latexindent` would stop `texlive` from adding `latexindent` to its `bin` folder, but it does not, and there does not seem to be an option to disable it anywhere.  The version `latexindent.pl` that comes with `texlive` is `3.9.2`, while the version in the `latexindent` formula is newer at `3.11`.  

We could delete the `libexec/bin/latexindent` or just the symlink to it in `bin/latexindent`.  That might be preferable to having a `conflicts_with` statement, as we'd force users to choose between using the older version when installing `texlive`, or the newer version that comes with the formula.

Reply:
Something that's not clear here though -- why are we installing into `libexec` rather than directly into `prefix`? Does this install non-library files into `lib`?

Reply:
I should have explained this more in the comments here, but the issue is that a lot of the scripts in TeXLive assume that the binaries are installed in an architecture subdirectory like `bin/x86_64-apple-darwin20.6.0`.  When we set `--disable-native-texlive-build`, by default this sets `--disable-multiplatform`, which disables the use of the architecture subdirectory.  

However, there are quite a few Perl scripts which are hardcoded to assume they are in a subdirectory of `bin` and don't know if the architecture subdirectory has been disabled.  We can re-enable the architecture subdirectory and then symlink the binaries into `bin` and everything will work, but this makes the keg fail `brew audit` because it contains the architecture subdirectory.  

Using `libexec` bypasses these issues because the Perl scripts only care about the directory depth, not the names of the directories, so being `libexec/bin` for them is equivalent to `bin/x86_64-apple-darwin20.6.0`.  The reason to use `libexec` here was because installing everything there and then symlinking it to the prefix passes `brew audit`, while using the architecture subdirectory fails the audit.  If anyone can suggest a better way to deal with this I'm willing to try it, but this was the only way I've been able to build it so far where all the TeXLive scripts work and it passes the audit.

Reply:
While this is good enough for me to get it merged, can we at least make upstream issues about these perl scripts?

Reply:
That sounds like a great idea.  It sounds like they use a mailing list for bug reports: https://www.tug.org/pipermail/tex-live/, so we'll end up with a link the mailing list archive.  At the very least I'd like to know why some of these relative paths are hardcoded as that seems to be more an oversight than a deliberate design decision.

Reply:
Okay I’ll make sure we delete these files.  This isn’t the only formula where we have to delete conflicting files like this.

Reply:
@SMillerDev here is the link to the email thread that has been started with TeXLive upstream: https://www.tug.org/pipermail/tex-live/2021-September/047394.html.  @norbusan has expressed an interest in helping.  He linked to some patches that Debian uses to deal with some of these path issues.  They are probably more extensive than what we would need, but hopefully we can come up with something that works for us and may even be possible to upstream.